lib/iinChecker.js

```diff
@@ -1,148 +1,166 @@
-					iin = String( iin );
-
-					// First up. Make sure we are passing a number in.
-					if( _.isNaN( parseInt( iin ) ) ) {
-						throw new TypeError( this.options.messages.PARAMETER_IIN_IS_NOT_A_NUMBER );
-					}
-
-					// Now make sure that we are passing in 6 characters
-					if( iin.length !== 6 ) {
-						throw new TypeError( this.options.messages.PARAMETER_IIN_IS_NOT_LONG_ENOUGH );
-					}
-
-				}
-			}
-			if ( !_.isFunction( callback ) ) {
-				throw new TypeError( this.options.messages.PARAMETER_CALLBACK_NOT_FUNCTION );
-			}
-			this.makeRequest( iin, 0, callback );
-		} catch ( err ) {
-			// Return the err in the error callback, if not possible return it.
-			if ( callback ) {
-				callback( err );
-			} else {
-				return err;
-			}
-		}
-	};
-} )();
\ No newline at end of file
+  // get some stuff we need to support IIN Checker
+  var _ = require( 'lodash' );
+  var request = require( 'request' );
+  // Load all of our providers into an array that we can loop over.
+  var providers = require( '../configs/providers' );
+
+  var iinChecker = function( iinOptions ) {
+    // default options
+    this.options = {
+      language: 'en',
+      cache: false
+    };
+    // if there's any overriding options, blat them over the defaults
+    this.options = _.extend( this.options, iinOptions );
+
+    // Define our constants for comparison
+    this.nullValue = 'UNKNOWN'; /* Value incase services are offline or return null */
+    this.brands = {
+      VISA: 'VISA',
+      MASTERCARD: 'MASTERCARD',
+      AMEX: 'AMERICAN EXPRESS',
+      DISCOVER: 'DISCOVER',
+      DINERS: 'DINERS CLUB',
+      JCB: 'JCB',
+      MAESTRO: 'MAESTRO',
+      LASER: 'LASER',
+      UNKNOWN: this.nullValue
+    };
+    // DANKORT what do we do with this?
+    // CHINA UNION PAY
+    // SOLO
+    // ELECTRON ??? Don't support? Visa Electron
+
+    // DINERS has been added for RegEx, but both Robbin and BinList return these cards as DISCOVER
+    // TODO: Change this later to be consitent across RegEx and providers
+
+    this.types = {
+      DEBIT: 'DEBIT',
+      CREDIT: 'CREDIT',
+      UNKNOWN: this.nullValue
+    };
+
+    // now fetch the language settings based on the requested language
+    this.options.messages = require( './i18n/' + this.options.language ); // ISO 639‑1
+
+    this.makeRequest = function( iin, providerCount, callback ) {
+      var self = this;
+      var cardInfo;
+      var providerDetails = providers[providerCount];
+
+      request( { url: providerDetails.domain + providerDetails.path + iin, json: true }, function( error, response, body ) {
+        if ( !error && response.statusCode === 200 ) {
+          // Remap the reply to match our schema
+          cardInfo = providerDetails.map( body, self.nullValue );
+          // if caching is on, provider request is only attempted if card details were not
+          // in the cache now we have the details set it.
+          if ( self.options.cache &&
```

Comment:
this is optional, maybe one for the future but would be nice to shard this if logic into a separate function, maybe `isCacheable` or something.  Would make this easier to read and test.